### ISSUE TITLE
Do we want a QC check to ensure that all deprecated classes have an obsoletion reason?

### ISSUE THREAD

**Issue by matentzn** [monarch-initiative/mondo#6968]
If we decide against it, just close this issue here.

**Comment by nicolevasilevsky** [comment]
it would be nice to have this but probably more effort than it is worth to go back and add obs reasons to all of the terms that do not already have one

**Comment by matentzn** [comment]
One option is to give all the ones that are currently missing an obsolescence reason a blanket reason (say, "out of scope") or similar, so that we can activate the QC check for the future. We could also create a (closed) issue explaining the obsoletion "Deprecated classes without an obsoletion reason" and link all classes to this ticket; this ticket could then have an explanation that we did not track obsoletion reasons when this class was obsoleted, but if you want to know why it was obsoleted, you can open a new ticket?

**Comment by nicolevasilevsky** [comment]
I like both of these ideas @matentzn. I'm in favor of doing this.

**Comment by twhetzel** [comment]
I am in favor of adding the qc check and also adding the blanket reason with a ticket reference for any class that does not already have an obsoletion reason.

**Comment by matentzn** [comment]
Yes, I didnt think to also say this clearly: Do we think that _all_ obsoleted classes should be linked to a GitHub issue? I think this would be a pretty awesome show of transparency as well.

**Comment by sabrinatoro** [comment]
Yes to everything BUT I think there are a lot of "old" obsoleted classes for which there might not be a GH issue. So we need to have a plan for these.

**Comment by matentzn** [comment]
The same idea as above: a blanket issue explaining that this term was obsoleted before there was a system, and there is therefore no issue tracker connected.

**Comment by matentzn** [comment]
@nicolevasilevsky can you create a beautifully formatted issue then, that explains this situation, and close it, and link it here? I will work on the rest.
